### PR TITLE
network: introduce Bare UDP Tunnelling

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

File: man/systemd.netdev.xml [hunk 1]
```diff
@@ -183,6 +183,8 @@
           <row><entry><varname>ifb</varname></entry>
           <entry> The Intermediate Functional Block (ifb) pseudo network interface acts as a QoS concentrator for multiple different sources of traffic.</entry></row>
 
+          <row><entry><varname>bareudp</varname></entry>
+          <entry> The Bare UDP tunnel module provides a generic L3 encapsulation support for tunnelling different L3 protocols like MPLS, IP etc inside an UDP tunnel.</entry></row>
```
Comment (poettering) [inline man/systemd.netdev.xml]: "The Bare UDP tunnel module provides a" → "Bare UDP tunnels provide"

inside an → inside of an

etc → etc.

File: man/systemd.netdev.xml [hunk 2]
```diff
@@ -822,6 +824,31 @@
...
+
+    <para>The [BareUDP] section only applies for
+    netdevs of kind <literal>bareudp</literal>, and accepts the
+    following keys:</para>
+
+    <variablelist class='network-directives'>
+      <varlistentry>
+        <term><varname>DestinationPort=</varname></term>
+        <listitem>
+          <para>Specifies destination port. This is mandatory. Default to unset.</para>
```
Comment (poettering) [inline man/systemd.netdev.xml]: specify range. maybe mention again this is an UDP port, here, i.e. `Specifies the destination UDP port (in range 1…65535).`

File: man/systemd.netdev.xml [hunk 3]
```diff
@@ -822,6 +824,31 @@
...
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>EtherType=</varname></term>
+        <listitem>
+          <para>Specifies the L3 protocol. Takes one of <literal>ipv4</literal>, <literal>ipv6</literal>, <literal>mpls-uc</literal>
+            or <literal>mpls-mc</literal>. This is mandatory. Default to unset.</para>
```
Comment (poettering) [inline man/systemd.netdev.xml]: defaults to unset if it's mandatory, how can that be?

File: man/systemd.netdev.xml (hunk 2 above)
Comment (poettering) [reply man/systemd.netdev.xml]: defaults to unset when mandatory? It doesn't default to anything, i.e. drop the "Default to unset"

File: src/network/netdev/bareudp.c [hunk 4]
```diff
@@ -0,0 +1,139 @@
+/* SPDX-License-Identifier: LGPL-2.1+
+ * Copyright © 2020 VMware, Inc. */
```
Comment (poettering) [inline src/network/netdev/bareudp.c:2]: are the explicitly Copyright lines required? We tried hard to remove them from our tree, would suck having to add them here again.

**Review by poettering — CHANGES_REQUESTED** [review]
Looks pretty OK to me, but I guess @yuwata should bless this.

File: man/systemd.netdev.xml (hunk 2 above)
Comment (ssahani) [reply man/systemd.netdev.xml]: opps yeah sorry

File: src/network/netdev/bareudp.c (hunk 4 above)
Comment (ssahani) [reply src/network/netdev/bareudp.c:2]: hmm yes

**Comment by yuwata** [comment]
Looks OK. I will add a test case for this.

File: man/systemd.netdev.xml [hunk 5]
```diff
@@ -822,6 +824,31 @@
...
+
+    <para>The [BareUDP] section only applies for
+    netdevs of kind <literal>bareudp</literal>, and accepts the
+    following keys:</para>
+
+    <variablelist class='network-directives'>
+      <varlistentry>
+        <term><varname>DestinationPort=</varname></term>
+        <listitem>
+          <para>Specifies the destination UDP port (in range 1…65535). This is mandatory</para>
```
Comment (poettering) [inline man/systemd.netdev.xml]: full stop at the end please.

File: man/systemd.netdev.xml [hunk 6]
```diff
@@ -183,6 +183,8 @@
           <row><entry><varname>ifb</varname></entry>
           <entry> The Intermediate Functional Block (ifb) pseudo network interface acts as a QoS concentrator for multiple different sources of traffic.</entry></row>
 
+          <row><entry><varname>bareudp</varname></entry>
+          <entry> The Bare UDP tunnels provide a generic L3 encapsulation support for tunnelling different L3 protocols like MPLS, IP etc. inside of an UDP tunnel.</entry></row>
```
Comment (poettering) [inline man/systemd.netdev.xml]: i'd drop the leading "The"

**Review by poettering — CHANGES_REQUESTED** [review]

**Comment by ssahani** [comment]
Updated thanks for the review.

**Comment by poettering** [comment]
lgtm, but I'll let @yuwata merge this

**Comment by ssahani** [comment]
Ping.

**Comment by yuwata** [comment]
@ssahani Sorry for late. Now I am testing this PR.

**Comment by yuwata** [comment]
@ssahani Tested. It seems to work as expected. Let's wait for CIs.

**Comment by ssahani** [comment]
@yuwata  thank you.

**Pull request merged**